src/ir/memory-utils.h

```diff
@@ -54,25 +54,23 @@ inline bool ensureLimitedSegments(Module& module) {
     return false;
   }
 
-  auto isEmpty = [](Memory::Segment& segment) {
-    return segment.data.size() == 0;
-  };
+  auto isEmpty = [](DataSegment* segment) { return segment->data.size() == 0; };
 
-  auto isConstantOffset = [](Memory::Segment& segment) {
-    return segment.offset && segment.offset->is<Const>();
+  auto isConstantOffset = [](DataSegment* segment) {
+    return segment->offset && segment->offset->is<Const>();
   };
 
   Index numConstant = 0, numDynamic = 0;
   bool hasPassiveSegments = false;
-  for (auto& segment : memory.segments) {
-    if (!isEmpty(segment)) {
-      if (isConstantOffset(segment)) {
+  for (auto& segment : dataSegments) {
+    if (!isEmpty(segment.get())) {
+      if (isConstantOffset(segment.get())) {
```

Comment:
We could use `*segment` instead of `segment.get()` here as well.

Reply:
Thanks! Updated all of the .get() spots I added in this file.